### PR TITLE
ISSUE-100: Add lazy loading of modules

### LINKED ISSUE #100: Update application to use lazy route loading
We need a means of demonstrating lazy loading within this application. I'd rather not pollute this application with unnecessary functionality but we need to either include lazy loading in the application or find another way to demonstrate how this can be implemented. I am thinking that this is either code in the app or a slide?

### LINKED ISSUE #124: Add side menu
Add a side menu and put all navigation into it.

### PR DESCRIPTION
**Pull request by mcacek**
Add lazy loading for modules with the exception of timesheet due to an unresolved issue. Upgrade zone.js to fix weird testing messages. Also ad sidenav.

Resolves #100.
Resolves #124.

### CODE REVIEW

File: ng-client/package.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
     "hammerjs": "2.0.8",
     "moment": "2.17.1",
     "rxjs": "5.1.0",
-    "zone.js": "0.8.4"
+    "zone.js": "0.8.5"
```
Comment (mcacek) [inline ng-client/package.json]: We were apparently using an old version of this.. 👎

File: ng-client/src/app/core/core.module.ts [hunk 2]
```diff
@@ -2,30 +2,45 @@ import {
...
+    HttpModule,
   ],
   providers: [
     ExtHttp,
     IdentityService,
     LocalStorage,
     LoginService,
     ResponseHandler,
+    AuthGuard,
+    ProjectService,
```
Comment (mcacek) [inline ng-client/src/app/core/core.module.ts]: ProjectService is used outside of the project module.

File: ng-client/src/app/core/login.component.ts [hunk 3]
```diff
@@ -13,9 +13,8 @@ import {
 } from '../core';
 
 @Component({
-  selector: 'at-login',
```
Comment (mcacek) [inline ng-client/src/app/core/login.component.ts]: I removed a bunch of selectors that don't do anything. They're routed directly via routers.

File: ng-client/src/styles.scss [hunk 4]
```diff
@@ -8,9 +8,6 @@ body {
 }
 
 .main-container {
-  margin-left: auto;
-  margin-right: auto;
-  max-width: 1024px;
```
Comment (mcacek) [inline ng-client/src/styles.scss]: This ended up looking pretty weird with navigation. I'm comfortable saying that we're targeting mobile devices due to our adoption of material.

File: ng-client/src/app/app.module.ts [hunk 5]
```diff
@@ -41,24 +30,17 @@ const extHttpConfig = {
 @NgModule({
   imports: [
     BrowserModule,
-    MaterialModule,
-    HttpModule,
-    FormsModule,
     CoreModule.forRoot(extHttpConfig),
-    EmployeeModule,
-    ProjectModule,
     TimesheetModule,
```
Comment (mcacek) [inline ng-client/src/app/app.module.ts]: I basically just gave up on this for now and included the timesheet module eagerly.

File: ng-client/package.json (hunk 1 above)
Comment (DSchau) [inline ng-client/package.json]: Always pretty cool bumping deps, but was this why things were breaking?

File: ng-client/src/app/app.module.ts [hunk 6]
```diff
@@ -1,29 +1,18 @@
 import { BrowserModule } from '@angular/platform-browser';
-import { NgModule, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
-import { HttpModule } from '@angular/http';
-import { FormsModule } from '@angular/forms';
+import { NgModule } from '@angular/core';
```
Comment (DSchau) [inline ng-client/src/app/app.module.ts]: I was wondering what that CUSTOM_ELEMENTS_SCHEMA was even doing. I think that's for Polymer elements or something :/

File: ng-client/src/app/core/core.module.ts [hunk 7]
```diff
@@ -2,30 +2,45 @@ import {
...
 
+import 'hammerjs';
+
 import { IdentityService } from './identity.service';
 import { LocalStorage } from './localStorage';
 import { LoginService } from './login.service';
 import { ResponseHandler } from './responseHandler.service';
 import { ExtHttp } from './extHttp.service';
 import { ExtHttpConfig } from './ExtHttpConfig';
+import { AuthGuard } from './auth-guard.service';
```
Comment (DSchau) [inline ng-client/src/app/core/core.module.ts]: Nice!

File: ng-client/package.json (hunk 1 above)
Comment (mcacek) [inline ng-client/package.json]: This is about all that I can give you. https://github.com/angular/angular/issues/15185 I believe it has to do with their approach to overriding the global Error object - https://github.com/angular/zone.js/blob/v0.8.5/CHANGELOG.md

File: ng-client/src/app/app.module.ts (hunk 6 above)
Comment (mcacek) [inline ng-client/src/app/app.module.ts]: Yeah, i'm looking at cleaning that up as well. This has to do with how we're including the angular material modules. If we include them correctly, we'll get rid of these errors.

**Pull request merged**